testbed/addons/flux/construct.ts

```diff
@@ -0,0 +1,85 @@
+import cdk = require('@aws-cdk/core');
+import eks = require('@aws-cdk/aws-eks');
+import * as yaml from 'js-yaml';
+import * as request from 'sync-request';
+import { KubernetesManifest } from '@aws-cdk/aws-eks';
+
+export interface FluxProps extends cdk.StackProps {
+    cluster: eks.Cluster;
+    repositories: Repository[];
+}
+
+export interface RepositoryOptions {
+    branch?: string;
+    path?: string;
+    url: string;
+    name: string;
+}
+
+export class Repository {
+    branch: string;
+    path: string;
+    url: string;
+    name: string;
+    constructor(options: RepositoryOptions) {
+        this.branch = options.branch ?? "main"
+        this.path = options.path ?? "test/workflows"
+        this.url = options.url
+        this.name = options.name
+    }
+}
+
+export class Flux extends cdk.Construct {
+    constructor(scope: cdk.Construct, id: string, props: FluxProps) {
+        super(scope, id);
+
+        // Actually install Flux components onto the cluster
```

Comment:
I'd avoid words like "actually".

In general I like to name things such that documentation isn't necessary (i.e. self documenting code). However, sometimes this isn't possible given the names of other components. In these cases, consider a documentation pattern of 

```
// 1. Do the first step
... 
// 2. Do another step
....
// 3. Do this one as well
...
```